no-undef misleading example

<!--
    ESLint adheres to the [JS Foundation Code of Conduct](https://js.foundation/community/code-of-conduct).

    This template is for requesting a change that is not a bug fix, rule change, or new rule. If you are here for another reason, please see below:

    1. To report a bug: https://eslint.org/docs/developer-guide/contributing/reporting-bugs
    2. To request a rule change: https://eslint.org/docs/developer-guide/contributing/rule-changes
    3. To propose a new rule: https://eslint.org/docs/developer-guide/contributing/new-rules
    4. If you have any questions, please stop by our chatroom: https://gitter.im/eslint/eslint

    Note that leaving sections blank will make it difficult for us to troubleshoot and we may have to close the issue.
-->

**The version of ESLint you are using.**

master

**The problem you want to solve.**

From `no-undef` docs:

> Examples of **correct** code for this rule with `global` declaration:
>
> ```js
> /*global someFunction b:writable*/
> /*eslint no-undef: "error"*/
>
> var a = someFunction();
> b = 10;
> ```
>
> The `b:writable` syntax in `/*global */` indicates that assignment to `b` is correct.
>
> Examples of **incorrect** code for this rule with `global` declaration:
> 
> ```js
> /*global b*/
> /*eslint no-undef: "error"*/
> 
> b = 10;
> ```
> 
> By default, variables declared in `/*global */` are read-only, therefore assignment is incorrect.

'Incorrect' examples should be those with warnings, but ESLint does not report any warnings for the second example.

The problem with this is that the second example is actually **correct**, because `no-undef` rule does not check assignments to readonly global variables, that's what `no-global-assign` does.

**Your take on the correct solution to problem.**

I'm not sure. Probably modify the first example and remove the second. Maybe reference `no-global-assign`.  Perhaps also add more info to  `no-global-assign`. 

**Are you willing to submit a pull request to implement this change?**
Yes.

The interesting this here is that `no-implicit-globals` also reports this, albeit in a different context.

```js
/* eslint no-implicit-globals:"error" */
/* eslint no-global-assign:"error" */
/* global a:readonly */

a = 1;
```

```
  5:1  error  Implicit global variable, assign as global property instead  no-implicit-globals
  5:1  error  Read-only global 'a' should not be modified                  no-global-assign
```

None of the rules will report anything if the global variable is set to `writeable`.

Is this the intended behavior of `no-implicit-globals`?

--- a/docs/rules/no-undef.md
+++ b/docs/rules/no-undef.md
@@ -25,6 +25,12 @@ var foo = someFunction();
 var bar = a + 1;
 ```
 
+Note that this rule does not disallow assignments to read-only global variables.
+See [no-global-assign](no-global-assign.md) if you also want to disallow those assignments.
+
+This rule also does not disallow redeclarations of global variables.
+See [no-redeclare](no-redeclare.md) if you also want to disallow those redeclarations.
+
 ## Options
 
 * `typeof` set to true will warn for variables used inside typeof check (Default false).
@@ -98,3 +104,8 @@ If explicit declaration of global variables is not to your taste.
 ## Compatibility
 
 This rule provides compatibility with treatment of global variables in [JSHint](http://jshint.com/) and [JSLint](http://www.jslint.com).
+
+## Related Rules
+
+* [no-global-assign](no-global-assign.md)
+* [no-redeclare](no-redeclare.md)

--- a/docs/rules/no-undef.md
+++ b/docs/rules/no-undef.md
@@ -11,32 +11,19 @@ Examples of **incorrect** code for this rule:
 ```js
 /*eslint no-undef: "error"*/
 
-var a = someFunction();
-b = 10;
+var foo = someFunction();
+var bar = a + 1;
 ```
 
 Examples of **correct** code for this rule with `global` declaration:
 
 ```js
-/*global someFunction b:writable*/
+/*global someFunction, a*/
 /*eslint no-undef: "error"*/
 
-var a = someFunction();
-b = 10;
+var foo = someFunction();
+var bar = a + 1;
 ```
-
-The `b:writable` syntax in `/*global */` indicates that assignment to `b` is correct.
-
-Examples of **incorrect** code for this rule with `global` declaration:
-
-```js
-/*global b*/
-/*eslint no-undef: "error"*/
-
-b = 10;
-```
-
-By default, variables declared in `/*global */` are read-only, therefore assignment is incorrect.
 
 ## Options
 

